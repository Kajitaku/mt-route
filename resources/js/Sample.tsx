import React, { useState } from "react";

type Props = {
  labelOn: string;
  labelOff: string;
};
const Sample = (props: Props) => {
  const [isChecked, setIsChecked] = useState(false);
  const { labelOn, labelOff } = props;
  return (
    <label htmlFor="sample">
      <input
        id="sample"
        type="checkbox"
        checked={isChecked}
        onChange={() => setIsChecked(!isChecked)}
      />
      {isChecked ? labelOn : labelOff}
    </label>
  );
};

export default Sample;
