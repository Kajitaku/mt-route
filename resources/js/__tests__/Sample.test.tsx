import React from "react";
import { shallow } from "enzyme";
import Sample from "../Sample";

test("Sample changes the text after click", () => {
  const checkbox = shallow(<Sample labelOn="On" labelOff="Off" />);

  // Interaction demo
  expect(checkbox.text()).toEqual("Off");
  checkbox.find("input").simulate("change");
  expect(checkbox.text()).toEqual("On");

  // Snapshot demo
  expect(checkbox).toMatchSnapshot();
});
