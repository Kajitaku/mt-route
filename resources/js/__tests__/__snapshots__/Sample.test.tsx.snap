// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample changes the text after click 1`] = `
<label
  htmlFor="sample"
>
  <input
    checked={true}
    id="sample"
    onChange={[Function]}
    type="checkbox"
  />
  On
</label>
`;
